services-camel/src/main/java/edu/unc/lib/dl/services/camel/fulltext/FulltextProcessor.java

```diff
@@ -105,19 +105,21 @@ public void process(Exchange exchange) throws Exception {
             return;
         }
 
-        Path derivativePath = Paths.get(derivativeBasePath, binarySubPath, binaryId + ".txt");
-        File derivative = derivativePath.toFile();
-        File parentDir = derivative.getParentFile();
-
-        // Create missing parent directories if necessary
-        if (parentDir != null) {
-            try {
-                Files.createDirectories(parentDir.toPath());
-            } catch (IOException e) {
-                throw new IOException("Failed to create parent directories for " + derivativePath + ".", e);
+        if (text.length() > 0) {
```

Comment:
I think we still want to write out the extract file even if its empty, since I think this is how we determine if a service should be re-run with the "Run Enhancements" action if the force flag isn't set to true. So I'd suggest removing this `if`